### DISCUSSION TITLE
Uhrzeiteinstellung und Sicherheitsschlüssel können nicht geschrieben werden

### DISCUSSION THREAD

**Discussion by Eldarion85** [OpenCCU/OpenCCU#2412] [🇩🇪 Fragen&Antworten] [answered]
### Describe the issue you are experiencing

Hallo,

ich habe ein komplett neues Proxmox 8.0.4 aufgesetzt und darin RaspberryMatic 3,71,12,20230826.
Das System läuft und das mit der eigenen Zertifikatskette klappt nach dem Update von 3.69 auf 3.71 auch wieder wunderbar.

Allerdings habe ich festgestellt, dass ich die Uhrzeit nicht setzen kann. Wenn ich das manuell versuche kommt die Meldung:
![Uhrzeit](https://github.com/jens-maus/RaspberryMatic/assets/90176159/5180b5ea-aada-444a-9d0e-20d39be72f56)

Ich nutze mein Synology als Zeitserver und darüber scheint es auch nicht zu gehen, denn in den Alarmmeldungen kommt sporadisch die Meldung von Chrony:
![Chrony](https://github.com/jens-maus/RaspberryMatic/assets/90176159/5a352615-d50a-4fee-86ad-2b233ff8fb53)

Außerdem kann ich den Sicherheitsschlüssel nicht setzen, denn dann kommt folgende Meldung:
![Schluessel](https://github.com/jens-maus/RaspberryMatic/assets/90176159/54f6f307-eae5-462b-9f0c-9f47b7a60018)

Dort steht zwar, dass ich Informationen unter den Servicemeldungen finde, aber es werden keine Servicemeldungen angezeigt.
Ob das mit der Uhrzeit und dem Sicherheitsschlüssel zusammenhängt weiß ich noch nicht.

### Describe the behavior you expected

Mein aktuell laufendes System ist Proxmox 7.0.4 und darin RaspberryMatic 3.67.10.20230114.
Das läuft seit 1,5 Jahren einwandfrei und dort geht auch die manuelle Zeiteinstellung, die Synchronisierung mit Synology und das Setzen des Sicherheitsschlüssel.

Gibt es eine Idee wo das Problem liegen könnte? Ich habe die VM auch nochmal komplett neu aufgesetzt und RaspberryMatic installiert, mit dem gleichen Ergebnis.

### Steps to reproduce the issue

1. Proxmox 8.0.4 installieren
2. In Proxmox Konsole einloggen
3. sudo installieren und Standardbenutzer in Gruppe sudo
4. RaspberryMatic-3.71.12.20230826-ova.zip runterladen und in Home-Verzeichnis von Standardbenutzer kopieren
5. RaspberryMatic-3.71.12.20230826-ova.zip entpacken
6. RaspberryMatic.ovf in Proxmox importieren (mit sudo ausführen)
7. Laufwerksgröße anpassen
8. Netzwerkkarte hinzufügen
9. OS Typ auf Linux 6.x - 2.6 Kernel umstellen
10. SCSI Controller auf VirtIO SCSI single umstellen
11. VM starten

### What is the version this bug report is based on?

3,71,12,20230826

### Which base platform are you running?

oci (Open Container Infrastructure)

### Which HomeMatic/homematicIP radio module are you using?

HmIP-RFUSB

### Anything in the logs that might be useful for us?

```txt
-
```

### Additional information

-

**Comment by jens-maus** [comment]
Also ich kann das Problem hier selbst nicht reproduzieren und TestVMs laufen hier problemlos und können auch die Uhrzeit problem einstellen ohne eine XmlRPC Fehlermeldung zu erhalten. Das muss also etwas lokales bei dir sein was hier das Problem verursacht.

Nach etwas durchsehen deines Tickets hier ist mir das hier aufgefallen:

> ich habe ein komplett neues Proxmox 8.0.4 aufgesetzt und darin eine VM mit Debian 12 und RaspberryMatic 3,71,12,20230826.

Das verstehe ich nicht. Was hast du da bitte gemacht? Du nutzt Proxmox 8 und hast da eine VM selbst aufgesetzt mit Debian 12 und darin dann RaspberryMatic als Docker Container installiert? Verstehe ich das richtig? Denn weiter unten beschreibst du jedoch die installation als "ova" installation um dann weiter unter wiederrum von "oci" zu sprechen. Das sind schon fundamentale unterschiede die es erst einmal zu klären gilt. Wie genau hast du jetzt bitte RaspberryMatic installiert und wie genau betreibst du es? Als vollwertige VM (ova) oder also Container variante (oci) unter einem anderen Host-Betriebssystem?

**Comment by Eldarion85** [comment]
Jetzt war die Antwort so schnell, dass ich meinen Beitrag nicht mehr editieren konnte :).
Sorry, ich sitze seit Tagen daran alle VM in Proxmox 8 neu aufzusetzen (Node-RED, ioBroker, AdGuard, etc.). Da kam ich etwas durcheinander. Ich habe natürlich keine Debian 12 VM erstellt, sondern lediglich in der Proxmox Konsole die RaspberryMatic.ovf importiert.

**Comment by jens-maus** [comment]
Warum verwendest du nicht erst einmal den offiziell dokumentierten Weg (siehe https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Proxmox-VE#installation) via einfachem `wget` Aufruf von der Proxmox Konsole aus? Das sollte dich schneller und effektiver zum Ergebnis führen und auch die Anbindung des HmIP-RFUSB dann entsprechend automatisch durchführen.

**Comment by Eldarion85** [comment]
Der zweite Weg steht ja auch in der Doku. Liegt an der Hardware-Firewall, dass ich diesmal den altarnativen Weg aus der Doku versucht habe. Bei meinem alten System was aktuell läuft hatte ich es mit wget gemacht.
Da ich nur einen HmIP-RFUSB und ein HMIP-HAP habe wollte ich die nachträglich einrichten. Denn die sind natürlich im alten System noch in Betrieb. Aber wenn es besser ist das direkt bei der Installation zu machen stecke ich die kurz um und danach wieder zurück, bis ich mit der Einrichtung fertig bin.

Mehr als die Datei zu entpacken und danach zu importieren ist es ja nicht.
Werde Proxmox in der Firewall mal freigeben und es jetzt nochmal mit
wget -qO - https://raspberrymatic.de/install-proxmox.sh | bash -
versuchen.

Sind denn meine Einstellungen der VM ok?
![Hardware](https://github.com/jens-maus/RaspberryMatic/assets/90176159/b91f2e01-639a-41ca-b812-46128401e4ca)

![Optionen](https://github.com/jens-maus/RaspberryMatic/assets/90176159/57f67a08-592f-4d4a-b075-9a30ee23c1c6)

**Comment by jens-maus** [comment, marked as answer]
> Der zweite Weg steht ja auch in der Doku. Liegt an der Hardware-Firewall, dass ich diesmal den altarnativen Weg aus der Doku versucht habe. Bei meinem alten System was aktuell läuft hatte ich es mit wget gemacht.

Verstehe ich. Ich kann jedoch nur dazu raten - wenn irgendwie möglich - auf die "wget"-Variante zu setzen. Da werden einfach schon viele Fallstricke einem abgenommen und auch die VM Einstellung optimaler angepasst als das in der manuellen/alternativen Installation so dokumentiert/beschrieben ist.

> Da ich nur einen HmIP-RFUSB und ein HMIP-HAP habe wollte ich die nachträglich einrichten. Denn die sind natürlich im alten System noch in Betrieb. Aber wenn es besser ist das direkt bei der Installation zu machen stecke ich die kurz um und danach wieder zurück, bis ich mit der Einrichtung fertig bin.

Ok, das ist ein wichtiges Detail und auch in deinen Screenshots sieht man das ja, das du das HmIP-RFUSB noch gar nicht zugewiesen hast. So geht das natürlich nicht. Eine CCU/RaspberryMatic braucht *immer* ein lokales Funkmodul egal ob du es produktiv hast oder nur testweise installiert. Das begründet auch den Fehler den du hier aufzeigst auch auch das mit Sicherheitsschlüssel! Dafür braucht es zwingend ein Funkmodul an der VM. Versteh aber ohnehin nicht warum du überhaupt da was "voreinstellen" willst wie datum usw.?!?! Sobald du das Backup (*.sbk) deiner alten aktuell laufenden installation nimmst und in die neue VM einspielst sind alle Einstellungen wie bei der alten VM rüberkopiert. D.h. du musst nicht irgendwelche NTP/Zeit Einstellungen vorher machen oder sonst was, denn die werden ohnehin wieder überschrieben werden vom sbk-Backup restore.

Da dies aber nun in der Tat kein Fehler/Issue von RaspberryMatic ist werd ich das mal in ein Diskussionsbeitrag konvertieren falls andere auf das gleiche Problem stoßen in Zukunft.

**Comment by Eldarion85** [comment]
Als ich meinen HMIP-HAP gekauft hatte wollte ich den HmIP-RFUSB eigentlich ganz ablösen und nur noch über Netzwerk arbeiten. Nachdem ich viele Beiträge gelesen habe ist mir bekannt, dass es ohne den HmIP-RFUSB am Ende trotzdem nicht geht, da der Stick irgendwelche notwendigen Aufgaben übernimmt.
Dass das Einstellen des Sicherheitsschlüssel deshalb nicht geht hatte ich vermutet, aber dass der Stick schon relevant ist wenn ich nur die Uhrzeit in der Software einstellen will, war mir noch nicht bekannt, Gut zu wissen.

Am einfachsten wäre es natürlich die ganze VM auf den neuen Proxmox zu verschieben. Dass ich das System neu aufsetze liegt nur daran, dass ich kein Fan von großen Betriebssystem-Upgrades bin, weil immer alter Müll zurück bleibt. Bei ioBroker z.B. installiere ich ein frisches Debian 12 und einen neuen ioBroker und spiele dann nur die Daten per Backup zurück. AdGuard das gleiche. Für Proxmox geht ja theoretisch auch der Befehl "pve7to8", wovon aber selbst die Doku von Proxmox abrät.
RaspberryMatic läuft wie ich gesehen habe ja unter Buildroot bzw. einem embedded Linix?!
![Homematic](https://github.com/jens-maus/RaspberryMatic/assets/90176159/077e0f6e-df6d-40b2-84bd-ff8cd94b1bba)

Ist das bei RaspberryMatic prinzipiell Unsinn und ich kann einfach die VM verschieben, oder ist es besser die neuste ovf zu importieren und dann darin einfach aus dem alten Backup das System wiederherzustellen?

Im Diskussionsbeitrag ist es jetzt mit den neuen Erkenntnissen definitiv besser aufgehoben.

**Reply by jens-maus** [reply]
> Am einfachsten wäre es natürlich die ganze VM auf den neuen Proxmox zu verschieben. Dass ich das System neu aufsetze liegt nur daran, dass ich kein Fan von großen Betriebssystem-Upgrades bin, weil immer alter Müll zurück bleibt. Bei ioBroker z.B. installiere ich ein frisches Debian 12 und einen neuen ioBroker und spiele dann nur die Daten per Backup zurück. AdGuard das gleiche. Für Proxmox geht ja theoretisch auch der Befehl "pve7to8", wovon aber selbst die Doku von Proxmox abrät.

Also davon weiss ich nichts denn beruflich haben wir vor kurzem das "pve7to8" auf einen ProxmoxCluster mit 23 Nodes losgelassen und haben keinerlei Probleme damit. Und persönlich halte diese 90er-Jahre Herangehensweise wie "Lieber frisch aufsetzen statt ein Upgrade zu machen"  im Bezug auf ein Linux-Betriebssystem vollkommen überholt. Und gerade im Bezug auf Proxmox ist das vollkommen unnötig und IMHO auch kontraproduktiv. 

> RaspberryMatic läuft wie ich gesehen habe ja unter Buildroot bzw. einem embedded Linix?! !
> Ist das bei RaspberryMatic prinzipiell Unsinn und ich kann einfach die VM verschieben, oder ist es besser die neuste ovf zu importieren und dann darin einfach aus dem alten Backup das System wiederherzustellen?

Ich halt das in der Tat auch für RaspberryMatic als vollkommen unsinnig hier die VM frisch aufsetzen zu wollen. Gerade da RaspberryMatic ja mit einem read-only rootfs daher kommt wird sich da während der nutzung absolut nichts drin ändern und alle Nutzerdaten liegen im userfs das ja komplett in den sbk-Backups liegt. Also ja, bei RaspberryMatic ist das von dir wohl präferierte "Lieber frisch aufsetzen statt Upgrade zu machen" vollkommen überzogen. Verschiebe die VM auf dein neues Proxmox, starte sie und fertig. Alles andere ergibt keinerlei Sinn und stiehlt dir nur Lebenszeit :)

**Comment by Eldarion85** [comment]
> Also davon weiss ich nichts denn beruflich haben wir vor kurzem das "pve7to8" auf einen ProxmoxCluster mit 23 Nodes losgelassen und haben keinerlei Probleme damit. Und persönlich halte diese 90er-Jahre Herangehensweise wie "Lieber frisch aufsetzen statt ein Upgrade zu machen" im Bezug auf ein Linux-Betriebssystem vollkommen überholt. Und gerade im Bezug auf Proxmox ist das vollkommen unnötig und IMHO auch kontraproduktiv.

Auf der Arbeit habe ich ebenfalls das Upgrade mit pve7to8 gemacht und hatte keine Probleme. Auch die Upgrades von Debian 11 auf 12 in den VM's liefen einwandfrei. Da ich selbst noch nie Probleme hatte, kann ich auch nicht wirklich begründen, warum ich privat bei den Debian und Proxmox Hauptversionen neu aufsetze :). Aber da das nur alle zwei Jahre der Fall ist hält sich der aufwand bei den 5-6 VM's privat ja auch in Grenzen.
Ich habe auch gerade in der Proxmox Doku geschaut. Der Punkt der da drin stand, dass eine Neuinstallation dem pve7to8 vorzuziehen ist, wurde in der Version von Juli entfernt :D.

> Ich halt das in der Tat auch für RaspberryMatic als vollkommen unsinnig hier die VM frisch aufsetzen zu wollen. Gerade da RaspberryMatic ja mit einem read-only rootfs daher kommt wird sich da während der nutzung absolut nichts drin ändern und alle Nutzerdaten liegen im userfs das ja komplett in den sbk-Backups liegt. Also ja, bei RaspberryMatic ist das von dir wohl präferierte "Lieber frisch aufsetzen statt Upgrade zu machen" vollkommen überzogen. Verschiebe die VM auf dein neues Proxmox, starte sie und fertig. Alles andere ergibt keinerlei Sinn und stiehlt dir nur Lebenszeit :)

OK, dann werden ich das bei RaspberryMatic so machen und einfach die VM verschieben. Besten Dank für die super schnelle Hilfe!
